audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java

```diff
@@ -15,43 +15,49 @@
  */
 package com.wultra.core.audit.base.util;
 
+import lombok.extern.slf4j.Slf4j;
+
 import java.util.List;
 
 /**
  * Utility for obtaining information about calling class.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ * @author Lubos Racansky, lubos.racansky@wultra.com
  */
-public class ClassUtil extends SecurityManager {
+@Slf4j
+public final class ClassUtil {
 
-    private static final ClassUtil INSTANCE = new ClassUtil();
+    private ClassUtil() {
+        throw new IllegalStateException("Should not be instantiated");
+    }
 
     /**
      * Get calling class.
      * @param packageFilter Packages to filter out when resolving calling class.
      * @return Calling class.
      */
-    public static Class<?> getCallingClass(final List<String> packageFilter) {
-        final Class<?>[] trace = INSTANCE.getClassContext();
-        if (trace == null) {
-            return null;
-        }
+    public static String getCallingClass(final List<String> packageFilter) {
+        final StackTraceElement[] trace = Thread.currentThread().getStackTrace();
 
-        for (final Class<?> t : trace) {
-            if (t.isAssignableFrom(ClassUtil.class)) {
+        for (final StackTraceElement t : trace) {
+            final String className = t.getClassName();
+            if (Thread.class.getName().equals(className) || ClassUtil.class.getName().equals(className)) {
```

Comment:
When logic is based on `String` and not `Class`, we get a little bit better results.
```
JhmTest.testAuditInfo  avgt    6  3,996 ± 0,031  us/op
3,996 ±(99.9%) 0,031 us/op [Average]
  (min, avg, max) = (3,984, 3,996, 4,011), stdev = 0,011
  CI (99.9%): [3,964, 4,027] (assumes normal distribution)
```

Reply:
Sure, let's use it.